pkg/kv/iterators.go

```diff
@@ -269,9 +271,11 @@ func (p *PartitionIterator) Next() bool {
 	if p.itr == nil {
 		p.itr, p.err = p.store.Scan(p.ctx, []byte(p.partitionKey), []byte(""))
 		if p.Err() != nil {
+			p.itrClosed = true
```

Comment:
already closed at this point - need to turn it off after the return

Reply:
Done